src/core/include/openvino/op/constant.hpp

```diff
@@ -518,13 +518,13 @@ class OPENVINO_API Constant : public Op {
         if (!std::is_same<T, StorageDataType>::value) {
             OPENVINO_ASSERT(!std::numeric_limits<T>::is_signed ||
                             std::numeric_limits<StorageDataType>::lowest() <= value);
-            OPENVINO_ASSERT(value <= std::numeric_limits<StorageDataType>::max());
+            OPENVINO_ASSERT(std::numeric_limits<StorageDataType>::max() >= value);
```

Comment:
should we enable `<=` operator for float16 ?

Reply:
Honestly it is already implemented :)